### PR TITLE
Read the Docs + CAG API docs

### PR DESCRIPTION
**Pull request by OmarIthawi**
This PR adds the following:

 - Sphinx docs hosted on: https://course-access-groups.readthedocs.io/
 - General introduction and misc. docs
 - The `/course-access-groups/` API endpoint (more to come)
 - A couple of sections that has no content yet
 - (Somewhat unrelated) a test case to ensure that `PATCH` actually works

### How to Review
Check the docs site https://course-access-groups.readthedocs.io/ and spot any issues.

#### Why the massive ping?

 - I pinged all Open edX engineers to let you know that we have something at Read The Docs.

### CODE REVIEW

File: docs/engineering_design.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.. _chapter-engineering_design:
+
+Engineering Design
+==================
+
+.. note::
+
+    This section is still work in progress.
```
Comment (johnbaldwin) [inline docs/engineering_design.rst]: "still a work in progress"

File: docs/rest_api.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+
+Course Access Groups
+--------------------
+
+These endpoints allow to create, edit and delete Course Access Group.
+
+List Groups
+~~~~~~~~~~~
+This endpoint returns a paginated list of json objects in "results".
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: Maybe JSON all in caps

File: docs/rest_api.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+Each object representing a single Course Access Group.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: "Each object represents"

File: docs/rest_api.rst [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+Group Details
+~~~~~~~~~~~~~
+This endpoint provides a detailed view of a single Course Access Group when
+providing an group identifier (``id`` for short).
+
+
+.. note::
+
+    At In this version this endpoint offers no additional information compared to
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: remove "At In"

File: docs/rest_api.rst [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+    {
+      "id": 2,
+      "name": "Sales Employees",
+      "description": "All team members from the sales team"
+    }
+
+Add a New Group
+~~~~~~~~~~~~~~~
+Performing a ``POST`` request to this endpoint allows to add a new group.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: "Perform a ``POST`` ...

File: docs/rest_api.rst [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+.. code-block:: bash
+
+    DELETE /course_access_groups/api/v1/course-access-groups/2/
+
+
+Linking Courses to Course Access Groups
+---------------------------------------
+
+These endpoints allow to add and remove courses from Course Access Groups.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: "These endpoints let us add and remove..."

File: docs/rest_api.rst [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+
+.. note::
+
+    This section is still work in progress.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: You could leave out "still"

File: docs/rest_api.rst [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+
+Marking Courses as Public
+-------------------------
+
+These endpoints allow making courses as a public course that is available to
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: "These endpoints allow us to make" or "These endpoints mark courses as public. This means the course is available to all site learners regardless of their..."

File: docs/rest_api.rst [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+all site learners regardless of their Course Access Groups membership.
+
+.. note::
+
+    This section is still work in progress.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: Can remove "still"

File: docs/rest_api.rst [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+
+
+
+User Membership in Course Access Groups
+---------------------------------------
+
+These endpoints allow to add and remove users from Course Access Groups.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: Same as previous comment, "These endpoints allow us to make" or "These endpoints mark courses as public. This means the course is available to all site learners regardless of their..."

File: docs/rest_api.rst [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+.. note::
+
+    This section is still work in progress.
+
+
+Rules for Automatic User Membership
+-----------------------------------
+
+These endpoints allow to manage rules for automatic membership based on
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: Same as previous comment, "These endpoints allow us to make" or "These endpoints mark courses as public. This means the course is available to all site learners regardless of their..."

File: docs/rest_api.rst [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+
+User Membership in Course Access Groups
+---------------------------------------
+
+These endpoints allow to add and remove users from Course Access Groups.
+
+
+.. note::
+
+    This section is still work in progress.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: can remove "still"

File: docs/rest_api.rst [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+.. note::
+
+    The membership rules are only activated after the learner (user)
+    activates their email address. Before that, the learner will be considered
+    as without a group.
+
+
+.. note::
+
+    This section is still work in progress.
```
Comment (johnbaldwin) [inline docs/rest_api.rst]: Can remove "still"

File: docs/user_stories.rst [hunk 14]
```diff
@@ -0,0 +1,10 @@
+.. _chapter-user_stories:
+
+User Stories
+============
+
+
+.. note::
+
+    Currently user stories are being drafted in our Confluence page. Please
```
Comment (johnbaldwin) [inline docs/user_stories.rst]: "Current user stories.."

**Comment by OmarIthawi** [comment]
Thanks a lot @johnbaldwin!

**Pull request merged**